### ISSUE TITLE
Transition Elements between magic move elements

### ISSUE THREAD

**Issue by niklas-wortmann** [slidevjs/slidev#1430]
**Is your feature request related to a problem? Please describe.**
I would like to have a changing title in my slide based on the code that is shown right now. Having a title that is in sync with magic move would be really nice to provide additional context and information to the code shown. Code surfer has a similar feature, demonstrated here: https://codesurfer.pomb.us/full/

**Describe the solution you'd like**
I think having a syntax like this makes sense (at least to me) and could utilize view transitions for the headlines : 
````md magic-move
# Service with a Subject
```angular-ts
@Component({
  standalone: true,
  template: `
     <div *ngFor="let item of source$ | async"></div>
  `,
})
export class FooComponent {
  source$ = inject(Service).someMethod();
}

```
# Service with a Signal

```angular-ts
@Component({
  standalone: true,
  template: `
     <div *ngFor="let item of source()"></div>
  `,
})
export class FooComponent {
  source = toSignal(inject(Service).someMethod());
}
```
````
**Describe alternatives you've considered**
Using the current click system could be confusing for users if something is considered a "new click" or in sync with the magic move clicks

**Comment by lucaslrodri** [comment]
I believe it would be beneficial to add a component called `MagicMove` that implenents smooth transitions between `slots`, taking into account all elements and their children (such as code blocks, paragraphs, equations, divs, lists, blockquotes, etc.), similar to the Auto-Animate feature in Reveal.js ([Auto-Animate](https://revealjs.com/auto-animate/)). For your use case, it could be implemented in this manner (using MDC):

````markdown
::MagicMove
#subject
# Service with a subject

```angular-ts
@Component({
  standalone: true,
  template: `
     <div *ngFor="let item of source$ | async"></div>
  `,
})
export class FooComponent {
  source$ = inject(Service).someMethod();
}
```
#signal
# Service with a Signal

```angular-ts
@Component({
  standalone: true,
  template: `
     <div *ngFor="let item of source()"></div>
  `,
})
export class FooComponent {
  source = toSignal(inject(Service).someMethod());
}
```
::
````

**Comment by KermanX** [comment]
Closes this and opened #1527, because #1527 is more universal. Thanks for your ideas!
